### DISCUSSION TITLE
Weiterer hardwareseitiger Tiefentladeschutz notwendig

### DISCUSSION THREAD

**Discussion by daltonthomas0815** [hoylabs/OpenDTU-OnBattery#661] [General]
Hallo zusammen,
habe auch mein Projekt fast abgeschlossen. 
HM600 mit 3x 12 V LifePO4 50Ah Batterien in Serie an OpenbatteryDTU. Die Batterien haben ein integriertes BMS inkl. Tiefentladeschutz.
Geladen wird die Batterie AC-Seitig mit einem Meanwell LED-Treiber, der über PWM mit einem Python Skript gesteuert wird.

Wie sind Eure Erfahrungen bzgl. Tiefentladeschutz der Batterien? Für den Fall dass der ESP gestört ist, würde ja die Entladung mit dem Wechselrichter nicht gestoppt. Denkt ihr, dass der Tiefentladeschutz über die BMS der Batterie ausreichend ist oder würdet ihr noch eine weitere hardwareseitige Schaltung empfehlen? Grüße Thomas

**Comment by spcqike** [comment]
Ich denke für den Fall, dass openDTU nicht läuft und den wechselrichter die Batterie leer saugt, ist das BMS die letzte reißleine. Dafür ist es ja theoretisch da. 

wenn du dem nicht vertraust, kannst du ein Batterie Wächter Relais einbauen. Welcher der auch immer, es muss halt den Strom dauerhaft leiten können. 

Eine weitere Idee wäre eine Art totmann-Steckdosen-Adapter. Ich stelle mir da bspw eine smarte Steckdose mit Tasmota vor, die auf einem MQTT topic der DTU reagiert und jedes Mal einen internen Timer zurück setzt. Wenn das zurücksetzen nicht klappt (openDTU geht nicht mehr, aber auch wlan Störung, MQTT broker Störung, …) würde die Steckdose AC seitig abschalten. 

Beide Ansätze haben ihre Vor- und Nachteile. 

Allerdings solltest du dir eher Gedanken machen, wie du deine 3x 12V in Reihe ausgleichen willst. Oder sind die BMS gekoppelt und in irgend einer Art und Weise passiert ein balancing?

**Reply by daltonthomas0815** [reply]
Hallo @spcqike, 
danke für die Idee mit dem Totmann-Steckdosen-Adapter.Das ist von der Verkabelung etc. eine elegante Sache und zieht nicht auch noch Strom von der Batterie ab. Ich war auf dem Ansatz einer Komparatorschaltung mit einem LM393 der die Batteriespannung misst und ein Relais (Öffnerkontakt an dem der Wechselrichter angeschlossen ist) betätigt.
Zu Beginn meines Projekts war ich mir sicher, dass ein weiterer Tiefentladeschutz ins System muss. Da openDU derzeit so stabil läuft (Danke an die Entwickler!) und alle Sicherheitssysteme seine Vor und Nachteile haben bin ich da gerade hin und hergerissen ob ich diesen Schutz noch einbaue oder auf die Reißleine BMS setze.

Jede 12 V Batterie hat sein eigenes BMS und Balancing ist ein Thema. Ich kann noch nicht abschätzen, wie die Batteriespannungen zeitlich driften. Für den Anfang beobachte ich die einzelnen Batteriespannungen und balance händisch. Wenn ich alle paar Monate eingreifen muss, ist das ok. Sollte es häufiger werden, dann überlege ich mir eine Lösung (neuer Speicher oder BMS).

**Reply by spcqike** [reply]
> Jede 12 V Batterie hat sein eigenes BMS und Balancing ist ein Thema. Ich kann noch nicht abschätzen, wie die Batteriespannungen zeitlich driften. Für den Anfang beobachte ich die einzelnen Batteriespannungen und balance händisch.

Dafür gibt es wohl solche balancer. Ich weiß nicht, ob diese auch 36V systemspannung machen könnten. Bei einigen älteren balancern für Blei war es so. 

https://www.ebay.de/itm/275766649426

im Zweifel brauchst du noch eine passende 12V Batterie und machst damit ein 48V System (wenn der laderegler und die Module das hergeben)

**Reply by daltonthomas0815** [reply]
@spcqike Danke für den Link, eine Schwierigkeit ist das Verhalten dieser BMS für den Fall dass eine Batterie über das interne BMS abschaltet. Das mögen diese Teile oft nicht.
Neben den Batterien müsste ich dann auch meinen Laderegler tauschen. Ich beobachte und lerne noch etwas mit meinem kleinen System (sind nur 3x50Ah Akkus) und wechsle dann später auf ein größeres 48 V Setup mit einem einzigen BMS....:).

**Comment by hoschiking** [comment]
Ich find,e das BMS sollte die letzte Instanz sein, die trennt und habe noch eine weitere Abschaltung vorgesehen.
Dafür gibt es diverse Ansätze, im Grunde genommen aber läuft es immer auf irgend eine Form eine spannungsgesteueerten Schalters.
Da gibt es diverse Möglichkeiten:
- Bei Verwendung eine Victron MPPT den Lastausgang nutzen, da ist direkt ein programmierbarer Tiefentladeschutz integriert.
   Damit kannst du dann irgendwas schalten, was den Wechselrichter trennt (Relais, Mosfet....)
- Schaltbarer Zwischenstecker mit Tasmota, den kannst du mittels MQTT schalten (ioBroker usw.....)
   Falls deine Steuerung die Batteriespannung kennt, ist es ein Einfaches den Wifi Zwischenstecker zu schalten.
   Falls dies nicht der Fall ist, müsstest du dir einen Spannungssensor an einen ESP anschlißen und damit auswerten. 

Denk dran, der Hoymiles wird DC-seitig am Leben gehalten, Das EIN/AUS sollte also am besten auf der AC-Seite erfolgen, dann bleibt er erreichbar. Des weiteren ist es kritisch DC-seitig zu schalten, der Einschaltstrom ist nicht ohne und ohne Begrenzung kann man den killen

**Reply by daltonthomas0815** [reply]
@spcqike , @hoschiking 
Habe jetzt den Totmannschalter mittels Tasmota-Steckdose umgesetzt und bin in Testphase.
Vorgehen: 
Tasmota-Rule startet Timer und schaltet Steckdose nach 10 Minuten ab.
Rule1 ON Power1#State=1 DO RuleTimer1 300 ENDON ON Rules#Timer=1 DO Power1 OFF ENDON

Über einen Http request von Nodered aus wird der Timer vor Ablauf zurückgesetzt.
http://192.168.178.132/cm?cmnd=RuleTimer1%20300

Nodered:
Es wird die Batteriespannungsmessung vom Hoymiles über Opendtu genutzt. Sollte die Spannung unter einen Schwellwert fallen oder die letzten 10 Minuten kein aktualisierter Messwert über MQTT eingegangen sein, dann wird das Senden des HTTP-requests unterbunden.
Besser wäre sicherlich noch eine autarke Batteriespannungsmessung, aber ich denke das sollte so auch funktionieren.

Grüße und Danke nochmals für den Tipp mit dem Totmannschalter!

**Reply by spcqike** [reply]
Schön dass du eine Lösung gefunden hast :) berichte gerne, wie sie läuft und ob alles klappt 

Kleine Anmerkung:
300s sind 5 Minuten ;) 

Und sollte es nicht möglich sein, mit tasmota auf ein MQTT topic zu lauschen? Kann die Steckdose nicht selbst auf die MQTT Nachricht mit der Batterie Spannung (oder Spannung vom DC inverter Eingang) reagieren? Ohne Umweg über node red? 

Ich vermute, node red läuft auf dem gleichen Gerät, wie dein MQTT Broker, dennoch sparst du so eine mögliche Fehlerquelle. 

Sollte deine node red Instanz bei einem Update oder sonst was stoppen, stoppt so auch dein inverter. 

Wenn du direkt auf die MQTT Nachricht reagieren lässt, wäre dieser Punkt halt unwichtig :) 

Zusätzliche Fehlerquellen gibt es noch genug (MQTT broker service, switch? Access point? Hardware vom server? ….)

**Reply by daltonthomas0815** [reply]
@spcqike: Das mit dem MQTT topic lauschen habe ich nicht ganz rund hinbekommen. Die MQTT Nachrichten sind angekommen, aber das zurücksetzen des Timers hat nicht geklappt. Auch ein paar Umwege, wie z.B. Rule1 triggert Rule2 usw. liefen nicht.
Nodered läuft auf dem gleichen Gerät wie der Broker, mal sehen wie stabil das ganze läuft. Ich habe mir mal die Logs der letzten 3 Monate angesehen und keine Unterbrechung festgestellt......ich werde nach einer Weile nochmals berichten.....Gute Anmerkung mit den 300s, 10 Min war der Plan...:) Aber zum testen bin ich auf 5 Minuten gegangen, damit ich nicht so lange warten muss..
